### PR TITLE
Fixed an issue where article list pages could throw an error for invalid filters

### LINKED ISSUE #4497: GenericFacetedListView not validating input (ValueError)
**Describe the bug**
A ValueError is being raised from ORM queries, rather than validating input on the /articles page

**Janeway version**
v1.7.2

**To Reproduce**
Alter an ID on a selection field to enter an invalid value (e.g. ["368'123".](https://open-library-of-humanities.sentry.io/issues/6031570558/?referrer=slack))

**Expected behavior**
Form should not validate and avoid sending invalid types via ORM

**Screenshots**

## Front-end Issues
If the issue is front-end specific please add the following details:

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by mauromsl**
The new article list views (Relying on GenericFacetedListView) rely on GET method to apply filtering. They use a django form to render facets and filters, however the form was not being validated, leading to server errors and potential SQL injection when malformed data was sent via query parameters. We now validate all query parameters ahead of executing any queries.

closes #4497

### CODE REVIEW

File: src/journal/urls.py [hunk 1]
```diff
@@ -65,12 +65,6 @@
         name='journal_articles',
     ),
 
-    re_path(
```
Comment (ajrbyers) [inline src/journal/urls.py:68]: This has leaked from another branch.

**Review by ajrbyers — CHANGES_REQUESTED** [review]
See comment.

**Review by ajrbyers — APPROVED** [review]
LOOKS GOOD TO ME!

🎖️🎖️🎖️🎖️🎖️🎖️🎖️🎖️🎖️🎖️🎖️🎖️🎖️

**Review by StephDriver — APPROVED** [review]
manually tested, works.

**Pull request merged**